hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpConfiguration.java

```diff
@@ -109,19 +110,26 @@ public ErrorHandler errorHandler() {
         /**
          * {@link ConnectionFactory} with enabled publisher confirms and
          * heartbeat.
-         * 
+         *
          * @param config
          *            with standard {@link RabbitProperties}
          * @return {@link ConnectionFactory}
+         * @throws GeneralSecurityException
+         *             in case of problems with enabled SSL connections
          */
         @Bean
-        public ConnectionFactory rabbitConnectionFactory(final RabbitProperties config) {
+        public ConnectionFactory rabbitConnectionFactory(final RabbitProperties config)
+                throws GeneralSecurityException {
```

Comment:
Why are you throwing a GeneralSecurityException and not a KeyManagementException,NoSuchAlgorithmException?